libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -547,97 +517,97 @@ void ESP8266WebServerTemplate<ServerType>::_streamFileCore(const size_t fileSize
 
 template <typename ServerType>
 const String& ESP8266WebServerTemplate<ServerType>::arg(const String& name) const {
-  for (int j = 0; j < _postArgsLen; ++j) {
-    if ( _postArgs[j].key == name )
-      return _postArgs[j].value;
-  }
-  for (int i = 0; i < _currentArgCount; ++i) {
-    if ( _currentArgs[i].key == name )
-      return _currentArgs[i].value;
-  }
-  return emptyString;
+    for (size_t j = 0; j < _postArgsLen; ++j) {
+        if ( _postArgs[j].key == name )
+            return _postArgs[j].value;
+    }
+    for (size_t i = 0; i < _currentArgCount; ++i) {
+        if ( _currentArgs[i].key == name )
+            return _currentArgs[i].value;
+    }
+    return emptyString;
 }
 
 template <typename ServerType>
-const String& ESP8266WebServerTemplate<ServerType>::arg(int i) const {
-  if (i >= 0 && i < _currentArgCount)
-    return _currentArgs[i].value;
-  return emptyString;
+const String& ESP8266WebServerTemplate<ServerType>::arg(size_t i) const {
+    if (i < _currentArgCount)
+        return _currentArgs[i].value;
+    return emptyString;
 }
 
 template <typename ServerType>
-const String& ESP8266WebServerTemplate<ServerType>::argName(int i) const {
-  if (i >= 0 && i < _currentArgCount)
-    return _currentArgs[i].key;
-  return emptyString;
+const String& ESP8266WebServerTemplate<ServerType>::argName(size_t i) const {
```

Comment:
Signature change

Reply:
it is fully source compatible because signed integers can always be numerically promoted to unsigned integers.